src/text_shaper/directwrite_shaper.cpp

```diff
@@ -448,88 +369,18 @@ void directwrite_shaper::shape(font_key _font,
     else {
         // Complex shaping
         const UINT32 textStart = 0;
-        DWriteAnalysisWrapper analysisWrapper(wText, d->userLocale);
+        dwrite_analysis_wrapper analysisWrapper(wText, d->userLocale);
 
         // Fallback analysis
-        ComPtr<IDWriteTextFormat> format;
-        auto hr = d->factory->CreateTextFormat(
-            fontInfo.description.wFamilyName.c_str(),
-            nullptr,
-            fontInfo.font->GetWeight(),
-            fontInfo.font->GetStyle(),
-            fontInfo.font->GetStretch(),
-            fontInfo.size.pt,
-            d->userLocale.c_str(),
-            &format);
-
-        ComPtr<IDWriteTextFormat1> format1;
-
-        if (SUCCEEDED(hr) && SUCCEEDED(format->QueryInterface(IID_PPV_ARGS(&format1))))
+        font_source_list sources = d->locator_->resolve(gsl::span(_text.data(), _text.size()));
```

Comment:
Note that only DWrite locator implements `resolve`. So basically you'll see zero font fallback for CJK if using DWrite shaper + Font config locator. 

Also I think resolve here need to also pass `fontDescription` for bold/italic/etc.

Reply:
I will catch up with them font config "resolve" implementation ASAP :)